### PR TITLE
[FIX] web: list view optional columns icon position

### PR DESCRIPTION
**Pull request by pparidans**
Before this commit, the column customization icon sometimes covers the
last column header in list views (or the 'ordering carret' icon).

Steps to reproduce:
- open Survey in list view
=> Avg Score column header overlaps with the icon

This commit adapts the headers last column's padding and the
customization icon's width to match each other and avoid overlapping.

It also adapts the vertical positionning of the icon and backport BS5 utility classes for `line-height`.

task-2816987

Related PR https://github.com/odoo/enterprise/pull/26974

### CODE REVIEW

File: addons/web/static/src/libs/bs5_utility_classes.scss [hunk 1]
```diff
@@ -93,3 +93,21 @@ $position-properties: (
...
+  sm: $line-height-sm,
+  base: $line-height-base,
+  lg: $line-height-lg,
+);
+
+@each $-value-key, $-value in $line-height-values {
+  .lh-#{$-value-key} {
+    line-height: #{$-value};
+  }
+}
```
Comment (pparidans) [inline addons/web/static/src/libs/bs5_utility_classes.scss:113]: @adr-odoo should I add all values or only the one used (and squash those commits)?

Comment (adr-odoo) [reply addons/web/static/src/libs/bs5_utility_classes.scss:113]: Putting all values is less confusing I think.

Comment (pparidans) [reply addons/web/static/src/libs/bs5_utility_classes.scss:113]: Or maybe only the one used (in this case `.lh-base`) ?

Comment (adr-odoo) [reply addons/web/static/src/libs/bs5_utility_classes.scss:113]: Putting all values is less confusing I think.

**Review by adr-odoo — APPROVED** [review]
@robodoo rebase-merge r+

**Pull request closed without merging**